src/renderer/components/kube-object/kube-object-list-layout.tsx

```diff
@@ -76,6 +77,15 @@ export class KubeObjectListLayout extends React.Component<KubeObjectListLayoutPr
         items={items}
         preloadStores={false} // loading handled in kubeWatchApi.subscribeStores()
         detailsItem={this.selectedItem}
+        customizeHeader={({ filters, ...placeholders }) => ({
+          filters: (
+            <>
+              {store.api.isNamespaced && <NamespaceSelectFilter />}
```

Comment:
at least it is more appropriate here than in ItemListLayout

Reply:
I agree but moving it here breaks other things and fixing those would probably be out of scope. So I am going to revert to a simpler fix, and then make a new PR to improve the customization of the header.